pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -498,8 +498,7 @@ protected boolean consumeLoop()
    * otherwise
    */
   private boolean processStreamEvents(MessageBatch messagesAndOffsets, long idlePipeSleepTimeMillis) {
-
-    int messageCount = messagesAndOffsets.getUnfilteredMessageCount();
+    int messageCount = messagesAndOffsets.getMessageCount();
     _rateLimiter.throttle(messageCount);
```

Comment:
@sajjad-moradi IMO the rate limiter should be applied to the filtered messages instead of unfiltered messages because the purpose of it is to limit the ingestion rate

Reply:
Since we don't provide any end.offset when we fetch messages, the only filtered out message type is the null message. Since we don't do any processing with those messages, it's ok to not count them towards rate limiting (although we've actually fetched them).